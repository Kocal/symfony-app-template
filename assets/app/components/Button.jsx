export default {
  name: 'SymfonyAppTemplateButton',
  props: {
    color: {
      type: String,
      default: 'primary',
    },
    size: {
      type: String,
      default: 'sm',
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    outline: {
      type: Boolean,
      default: false,
    },
    auto: {
      type: Boolean,
      default: false,
    },
    loading: {
      type: Boolean,
      default: false,
    },
    icon: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    genClasses() {
      return [
        'btn',
        `btn-${this.size}`,
        `btn-${this.color}`,
        this.loading && 'btn-loading',
        this.icon && 'btn-icon',
        this.disabled && 'disabled',
        this.outline && 'btn-outline',
        this.auto && 'btn-auto',
      ];
    },
  },
  render() {
    return (
      <button disabled={this.disabled} class={this.genClasses()} on={this.$listeners} {...this.$attrs}>
        {this.loading ? (
          <div class="loading">
            <i class="fa fa-spinner-third fa-spin" />
          </div>
        ) : (
          this.$slots.default
        )}
      </button>
    );
  },
};
