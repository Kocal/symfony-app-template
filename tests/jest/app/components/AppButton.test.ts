import { mount } from '@vue/test-utils';
import Button from '@app/components/AppButton.vue';

describe('AppButton', () => {
  it('default render', () => {
    expect.assertions(1);

    const wrapper = mount(Button);

    expect(wrapper.html()).toMatchInlineSnapshot(`<button class="btn btn-sm btn-primary"></button>`);
  });

  it.each(['xs', 'sm', 'lg', 'xl'])('size "%s"', (color) => {
    expect.assertions(1);

    const wrapper = mount(Button, {
      propsData: {
        color,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it.each(['primary', 'danger', 'success'])('color "%s"', (color) => {
    expect.assertions(1);

    const wrapper = mount(Button, {
      propsData: {
        color,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it.each(['icon', 'outline', 'auto'])('basic property "%s"', (property) => {
    expect.assertions(1);

    const wrapper = mount(Button, {
      propsData: {
        [property]: true,
      },
    });

    expect(wrapper.html()).toBe(`<button class="btn btn-sm btn-primary btn-${property}"></button>`);
  });

  it('disabled', () => {
    expect.assertions(1);

    const wrapper = mount(Button, {
      propsData: {
        disabled: true,
      },
    });

    expect(wrapper.html()).toMatchInlineSnapshot(
      `<button disabled="" class="btn btn-sm btn-primary disabled"></button>`
    );
  });

  it('loading', () => {
    expect.assertions(1);

    const wrapper = mount(Button, {
      propsData: {
        loading: true,
      },
    });

    expect(wrapper.html()).toMatchInlineSnapshot(`
      <button class="btn btn-sm btn-primary btn-loading">
        <div class="loading"><i class="fa fa-spinner-third fa-spin"></i></div>
      </button>
    `);
  });

  it('complex render', () => {
    expect.assertions(1);

    const wrapper = mount(Button, {
      propsData: {
        color: 'success',
        size: 'xl',
        outline: true,
        disabled: true,
        auto: true,
        icon: true,
        loading: true,
      },
    });

    expect(wrapper.html()).toMatchInlineSnapshot(`
      <button disabled="" class="btn btn-xl btn-success btn-loading btn-icon disabled btn-outline btn-auto">
        <div class="loading"><i class="fa fa-spinner-third fa-spin"></i></div>
      </button>
    `);
  });
});
