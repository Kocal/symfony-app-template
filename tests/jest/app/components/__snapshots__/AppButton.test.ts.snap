// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton color "danger" 1`] = `
<button
  class="btn btn-sm btn-danger"
>
  
  
</button>
`;

exports[`AppButton color "primary" 1`] = `
<button
  class="btn btn-sm btn-primary"
>
  
  
</button>
`;

exports[`AppButton color "success" 1`] = `
<button
  class="btn btn-sm btn-success"
>
  
  
</button>
`;

exports[`AppButton size "lg" 1`] = `
<button
  class="btn btn-sm btn-lg"
>
  
  
</button>
`;

exports[`AppButton size "sm" 1`] = `
<button
  class="btn btn-sm btn-sm"
>
  
  
</button>
`;

exports[`AppButton size "xl" 1`] = `
<button
  class="btn btn-sm btn-xl"
>
  
  
</button>
`;

exports[`AppButton size "xs" 1`] = `
<button
  class="btn btn-sm btn-xs"
>
  
  
</button>
`;
